src/bindings/python/flux/message.py

```diff
@@ -54,7 +55,7 @@ def handle(self):
     def from_event_encode(cls, topic, payload=None):
         if payload is None:
             payload = ffi.NULL
-        elif not isinstance(payload, basestring):
+        elif not isinstance(payload, six.text_type):
```

Comment:
Elsewhere I believe you used `six.binary_type`.  What is the procedure for determining which to use?

Reply:
That's actually a bad assumption on my part I didn't catch because I forgot to run the python2 test.  It matches unicode in py2 and str in py3.

It should probably be `six.string_types` for basestring in 2 and str in 3, json.dumps doesn't work on bytes objects, so we can't take them directly without encoding.  I'll fix that in a sec.

Reply:
As a more general answer, where it should take a str or bytes it should check against binary_type, where it should take a string of text, either string_types or text_type depending on whether we care about unicode in 2.